File: javax/src/main/java/me/panpf/javax/io/UnableCreateFileException.java
package me.panpf.javax.io;

import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.io.IOException;

public class UnableCreateFileException extends IOException {

    @NotNull
    public File file;

    public UnableCreateFileException(@NotNull File file) {
        super(file.getPath());
        this.file = file;
    }

    public UnableCreateFileException(@NotNull File file, @NotNull Throwable cause) {
        super(file.getPath(), cause);
        this.file = file;
    }
}

File: javax-kt/src/main/java/me/panpf/javaxkt/io/UnableCreateFileException.kt
package me.panpf.javaxkt.io

import java.io.File

class UnableCreateFileException : Exception {
    @Suppress("MemberVisibilityCanBePrivate")
    val file: File

    constructor(file: File) : super(file.path) {
        this.file = file
    }

    constructor(file: File, cause: Throwable) : super(file.path, cause) {
        this.file = file
    }
}